job-dsls/jobs/FlowJob.groovy

```diff
@@ -192,6 +196,88 @@ job("errai-kieAllBuild-${kieMainBranch}") {
 
 }
 
+// ++++++++++++++++++++++++++++++++++++++++++ Build and deploys kie-soup ++++++++++++++++++++++++++++++++++++++++++++++++++
+
+// definition of kiesoup script
+def kiesoupVersionBuild='''#!/bin/bash -e
+# removing kiesoup artifacts from local maven repo (basically all possible SNAPSHOTs)
+if [ -d $MAVEN_REPO_LOCAL ]; then
+rm -rf $MAVEN_REPO_LOCAL/org/kie/kie-soup
+       fi
+# clone the kiesoup repository
+git clone https://github.com/$organization/kie-soup.git -b $kiesoupBranch --depth 100
+# checkout the release branch
+cd kie-soup
+git checkout -b $kiesoupVersion $kiesoupBranch
+# update versions
+sh scripts/release/update-versions.sh $kiesoupVersion
+# build the repos & deploy into local dir (will be later copied into staging repo)
+deployDir=$WORKSPACE/deploy-dir
+# (1) do a full build, but deploy only into local dir
+# we will deploy into remote staging repo only once the whole build passed (to save time and bandwith)
+mvn -U -B -e clean deploy -T2 -Dfull -Drelease -DaltDeploymentRepository=local::default::file://$deployDir -s $SETTINGS_XML_FILE\\
+ -Dmaven.test.failure.ignore=true -Dgwt.compiler.localWorkers=3
+# (2) upload the content to remote staging repo
+cd $deployDir
+mvn -B -e org.sonatype.plugins:nexus-staging-maven-plugin:1.6.5:deploy-staged-repository -DnexusUrl=https://repository.jboss.org/nexus -DserverId=jboss-releases-repository -DrepositoryDirectory=$deployDir\\
+ -s $SETTINGS_XML_FILE -DstagingProfileId=15c3321d12936e -DstagingDescription="kie-soup $kiesoupVersion" -DstagingProgressTimeoutMinutes=30'''
```

Comment:
We should somehow make sure that we can have these scripts outside of the DSL code. This makes them untestable on its own.

Reply:
@psiroky I have on my to do list to refactor all this

Reply:
You can not expect to have an explicit time for refactoring. That basically never happens. Refactoring needs to be continuously practiced in small bursts, and of course accounted in the estimates. Anyway, I will merge the PR since that's probably the only choice at this point.